src/base/utils/os.cpp

```diff
@@ -283,45 +287,74 @@ bool Utils::OS::applyMarkOfTheWeb(const Path &file, const QString &url)
     // https://searchfox.org/mozilla-central/rev/ffdc4971dc18e1141cb2a90c2b0b776365650270/xpcom/io/CocoaFileUtils.mm#230
     // https://github.com/transmission/transmission/blob/f62f7427edb1fd5c430e0ef6956bbaa4f03ae597/macosx/Torrent.mm#L1945-L1955
 
+    const CFStringRef fileString = file.toString().toCFString();
+    [[maybe_unused]] const auto fileStringGuard = qScopeGuard([&fileString] { ::CFRelease(fileString); });
+    const CFURLRef fileURL = ::CFURLCreateWithFileSystemPath(kCFAllocatorDefault
+        , fileString, kCFURLPOSIXPathStyle, false);
+    [[maybe_unused]] const auto fileURLGuard = qScopeGuard([&fileURL] { ::CFRelease(fileURL); });
+
+    if (CFDictionaryRef currentProperties = nullptr;
+        ::CFURLCopyResourcePropertyForKey(fileURL, kCFURLQuarantinePropertiesKey, &currentProperties, NULL)
```

Comment:
Do you assume that the function can complete successfully (i.e. return true), but `currentProperties` will still be `nullptr`?

Reply:
I didn't 'assume' it. It happens.

Reply:
> I didn't 'assume' it. It happens.

It sounds confusing. So I even consulted the documentation. And it states that the function returns true if this parameter is successfully populated:
https://developer.apple.com/documentation/corefoundation/1542764-cfurlcopyresourcepropertyforkey#return_value

Reply:
You missed this part:
>If this function returns YES and the propertyValueTypeRefPtr is populated with nil, it means that the resource property is not available for the specified resource, and that no errors occurred when determining that the resource property was unavailable.